[UI] Layout of New Node Page

ChengYanJin opened the pull request:
**Component**: ui, nodes

**Context**: 
We want to set up the layout of empty node page, breadcrumb with empty node table only with Header (Health/Name/Role/Status), no node selected on the right side panel.

**Summary**:
Initialize the components for New Node Page:

`<NodePage>` fetch all the data needed for New Node Page and pass them to <NodePageContent/> component
`<NodePageContent>`
- Set up the layout of NodePage included Nodelist table and Right-side panel
- Retrieve the nodelist from <NodePage>
- Retrieve the current node from the URL to display the information in RSP
`<NodeListTable>`

**Acceptance criteria**: 
The **temporary URL** to access the New Node Page is: `/newNodes`
![image](https://user-images.githubusercontent.com/18453133/92460763-a1e34200-f1c8-11ea-8ecf-024f081df2cf.png)


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2771 

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ChengYanJin committed:
ui/nodes: Remove the comments in layout

It should have been cleaned up in New Volume Page PR

Refs: #2771

ChengYanJin committed:
ui/nodes: Add the temporary access to the New Node Page

Will change the URL to `/nodes` when all the work are done for New Node Page

Refs: #2771

ChengYanJin committed:
ui/nodes: Add platform path in the breadcrumb of Node Creation Page

Refs: #2771

ChengYanJin committed:
ui/translations: Add the translation of no node selected

Refs: #2771

ChengYanJin committed:
ui/nodes: Initialise the New Node Page Component

The <NodePage/> component is in charge of fetch all the data needed for New Node Page,
and pass them to <NodePageContent/> component

Refs: #2771

ChengYanJin committed:
ui/nodes: Initiaise the NodePageContent component

The <NodePageContent/> component
- Set up the layout of NodePage included Nodelist table and Right-side panel
- Retrieve the nodelist from <NodePage>
- Retrieve the current node from the URL to display the information in RSP

Refs: #2771

ChengYanJin committed:
ui/nodes: Initialise the NodeListTable component

Todo: Need to abstract the table component to reuse it
To be discussed: Do we want to have global search or smart search?

Refs: #2771

ChengYanJin committed:
ui/styles: Abstract the commont styles used by New Volume Page and New Node Page

Refs: #2771

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort

ChengYanJin committed:
ui/volumes: Use the abstracted common styles

Refs: #2771

ChengYanJin committed:
ui/volumes: Fix the mistake to extract the current volume name from URL in volume page

If there is no volume selected, the URL is `/volumes` so the currentVolumeName is `volumes`
which is wrong

Refs: #2771

thomasdanan reviewed (changes requested):
Tested it and works fine.
To be consistent with Volumes table, I suggest you have the following order for table Header:
Name, Role, Health, Status

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/2771-layout-of-new-node-page` and the
destination branch `development/2.6`.

Please resolve the conflict on **the feature branch** (`improvement/2771-layout-of-new-node-page`).

```bash
 $ git fetch
 $ git checkout origin/improvement/2771-layout-of-new-node-page
 $ git merge origin/development/2.6
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/2771-layout-of-new-node-page
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan

ChengYanJin commented:
> Tested it and works fine.
> To be consistent with Volumes table, I suggest you have the following order for table Header:
> Name, Role, Health, Status

Talked with Valamir about this topic before, he actually suggested that we have a little difference between Volume Page and Node Page on purpose. So that the user easily realizes that it's a different page.

But anyway, I've changed the order. We'll see later

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort

ChengYanJin commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



Please check the status of the associated issue None.

Goodbye chengyanjin.
